### ISSUE TITLE
[Explore]: Investigate kpack ClusterBuilder order property

### ISSUE THREAD

**Issue by gnovv** [cloudfoundry/cf-k8s-controllers#418]
### Background

As part of implementing the `cf buildpacks` command, we want to know what the behavior of the ClusterBuilder is when the order fields are not specified. We also want to know what fields of the `status` can be relied upon to be present.

### Acceptance Criteria

Note on a comment on this issue what the ordering behavior of a ClusterBuilder is if not specified explicitly. Report if there is a way of capturing this order via fields in the ClusterBuilder or related resources

### Number of days this exploration is expected to take. This is what is “pointed”.

1

**Comment by julian-hj** [comment]
**TL;DR:**  No `order` key in the status on ClusterBuilder is effectively the same as having no buildpacks.

We deleted the `order` key from the ClusterBuilder yaml in the `dependencies/kpack` directory and re-applied it. After doing that, we see that the ClusterBuilder CR becomes `ready` and has no `order` key in the `Status`.

However, when we push an app using this ClusterBuilder, it fails to stage. We got the following output, which suggests that the `order` is required by kpack to do detection:

```
prepare Build reason(s): CONFIG
prepare CONFIG:
prepare     resources: {}
prepare     - source: {}
prepare     + source:
prepare     +   registry:
prepare     +     image: localregistry-docker-registry.default.svc.cluster.local:30050/kpack/packages/d51d156b-e758-42d2-ab0b-070899411b55@sha256:579f6e9ef23258788347ec67a5bb980d53dad96ee8a3ff21ffb0e8f83358294b
prepare     +     imagePullSecrets:
prepare     +     - name: image-registry-credentials
prepare Loading secret for "localregistry-docker-registry.default.svc.cluster.local:30050" from secret "image-registry-credentials" at location "/var/build-secrets/image-registry-credentials"
prepare Pulling localregistry-docker-registry.default.svc.cluster.local:30050/kpack/packages/d51d156b-e758-42d2-ab0b-070899411b55@sha256:579f6e9ef23258788347ec67a5bb980d53dad96ee8a3ff21ffb0e8f83358294b...
prepare Successfully pulled localregistry-docker-registry.default.svc.cluster.local:30050/kpack/packages/d51d156b-e758-42d2-ab0b-070899411b55@sha256:579f6e9ef23258788347ec67a5bb980d53dad96ee8a3ff21ffb0e8f83358294b in path "/workspace"
analyze Previous image with name "localregistry-docker-registry.default.svc.cluster.local:30050/cf-k8s-controllers/kpack/images/1208efff-6022-4bac-9320-45fbd03edff4" not found
detect ERROR: No buildpack groups passed detection.
detect ERROR: Please check that you are running against the correct path.
detect ERROR: failed to detect: no buildpacks participating
```
Out of curiosity, we tried adding a manifest to the app with `buildpack: paketo-buildpacks/nodejs` added to it, but that didn't make any difference. It's quite possible that we just ignore that key in the manifest. 

Kpack docs for `order` are [here](https://github.com/pivotal/kpack/blob/main/docs/builders.md#order) but don't really say much about whether order is required.

**Comment by tcdowney** [comment]
These docs mention order is required. I'm surprised it even lets you make the `Builder` 😅 

https://buildpacks.io/docs/reference/config/builder-config/#order-_list-required_
